RevB

micade145 opened the pull request:
Added MCU and CAN components from psom; migrated to KiCad 9

Issues
* Do I need any of the power components from the psom?
* Couldn't find the DC-DC converter part (RO-123.3S_HP) in the library or parts folder; the one I added seems to have a different footprint

micade145 committed:
initial sch change, added PSOM components

micade145 committed:
migrated to KiCad 9

micade145 committed:
replaced UTSVT with KiCAD-Bild submodule

micade145 committed:
updated symbol, footprint paths

micade145 committed:
mcu and CAN added, missing Amperes DC-DC sym / footprint

micade145 committed:
added RO-123.3S_HP part but it might not be the right one

Rav4s commented:
for power stuff: the RO-123.3S_HP buck was giving us inaccurate voltage, so @Lakshay983 was thinking to switch to an isolated 12->5v buck and then 5->3.3v LDO. the buck on active precharge board should work (WE 1769205241). the buck on the rev of amperes we ordered could also work, @KadhirTheEngineer did you verify that one? also copying @ParthivS20 for sizing (mainly depth) constraints, this buck will prob be the tallest component on the board

Rav4s commented:
- lol i think you meant VDDA here (not 3.3v)
![image](https://github.com/user-attachments/assets/8cc62aed-b712-46a9-93d4-f589a0d0b3e4)
- did we wanna standardize SWD connector @Lakshay983 or pin headers are fine?
- if these are all no connect you can just remove the node
![image](https://github.com/user-attachments/assets/eb68952c-0ed4-4b30-ac4a-7af18a3e9261)
![image](https://github.com/user-attachments/assets/b49d2365-b503-4788-b796-421860c77a18)
- might want to break out a couple status LEDs?
- CAN looks good to me

micade145 committed:
addressed comments

micade145 committed:
isolated 3v3 done?

micade145 committed:
finished? power on sch

micade145 committed:
added MX cube pinout screenshot

micade145 committed:
power

micade145 commented:
Here's my progress so far
- Updated power: now using a 12V -> 5V Buck and 5V -> 3.3V LDO for generating both isolated and non-isolated 3.3V. A
- Addressed previous comments

In the works
- Still need to source a connector for SWD and some addressable LEDs that Lakshay asked me to add

Questions
- enough status LEDs or more?
- where is VDDA generated?

Rav4s commented:
looks great! a few comments:
- need to switch the buck to a 24 -> 5v (WE prolly makes one with the same footprint 👀)
- maybe add decoupling caps to the 24v input of the bucks 
<img width="346" alt="image" src="https://github.com/user-attachments/assets/f5c79be4-c24c-4eaa-9511-8b445cb40a53" />

- is this enough resistance for the 12v (now 24v) led?
<img width="445" alt="image" src="https://github.com/user-attachments/assets/10fcbc4a-8520-4636-ba98-c11ecb4e7612" />

- @Lakshay983 @FrankLiTX did we wanna switch swd to a standardized conn? 
<img width="305" alt="image" src="https://github.com/user-attachments/assets/7cbea816-7b14-4516-b53f-f087971735ed" />

- always good to add more LEDs, you have the pins lol and we can remove later if space constraints are that bad
- i think you should be good to move to layout, lmk if you have any questions about the sizing for the shunt & board dimensions
- also to answer your question, VDDA is generated by 3.3v, just separated by the ferrite bead to reduce noise from digital ICs interfering with analog. this is also why gnd and gnda are connected only at a single point (net-tie)

Lakshay983 reviewed (changes requested):
Add many bypass caps to your buck and power lines.  don't just add more of one value of capacitor. You want many value of capacitors since different values block different frequencies of noise better

Did you pick new SWD connector? @micade145 ? lmk if ur having trouble finding one. We just need one that's keyed and we can crimp a mating connector too

The INP and INN pin headers were more so for testing, but now we need to connect the actual shunt to it so you'll need to use bolts and mounting holes in order to connect to the board

we need to add a circuit on here for scrutineering, basically we just need an input into the board to spoof a current. easier to explain while talking, lmk when you want to talk about it.

micade145 committed:
added 24V buck, bypass caps, more LEDs; started scrutineering circuit and bidirectional current sense

micade145 committed:
organization, updated submodule

micade145 committed:
small tweaks

micade145 committed:
shunt to current sense to isolation done?? need to select values

micade145 committed:
new isolated amp, filter stage; need to select values

micade145 committed:
lowk got some filter values from chat PLEASE PLEASE PLEASE VERIFY LATER

micade145 committed:
plz review

micade145 commented:
Changes
- switched to 24V buck
- bidirectional current sense amp
- moved isolation after current sense
- low-pass filter

Still need to work out values for everything

micade145 committed:
comments

micade145 committed:
changed input into AMC1350, added unity gain buffer

Rav4s reviewed (changes requested):
great job, had some notes:

sch:
- nitpick but you're isolating across the buck so the line should go here
<img width="1109" height="607" alt="image" src="https://github.com/user-attachments/assets/f3c5bf98-295b-4d7a-b4fe-53241e113819" />

- use the standard LT1963AEST-3.3#TRPBF LDO for your 5v to 3.3v
- choose resistor values for your LEDs. Use XL-1608UBC-06A for blue and NCD0603R1 for red
- do we not want USB-C on this board? not enough space? would be good for debugging at least. if you add that make sure to add boot switch as well (CJS-1200TA) to select UART bootloader (copy psom schematic). and make sure UART/CAN pins are good for bootloader
- GND and GNDA should be connected with a Pi filter like you did for 3.3v and VDDA
- CAN looks good!
- can you add a little explanation of how scrutineering would work? refer to how lakshay did on volttemp sch
- do we have a footprint for the shunt?
- when you select gain, put this table in there with the one you're selecting + justification
<img width="1686" height="341" alt="image" src="https://github.com/user-attachments/assets/28030001-7104-455e-8304-c4f3e49a51ca" />

- on that note, do you think you can finish the math at workday? with @Lakshay983 
- why have ref1 and ref2 connected here? i might be reading the datasheet wrong but seems like ref1 is Vs and ref2 is GND.
<img width="1091" height="878" alt="image" src="https://github.com/user-attachments/assets/27a98940-61f5-4d4d-8ed4-6612791a6507" />

- need TVS diode for ADC input on MCU?

micade145 committed:
added LEDs, changed LDO, still need to address other comments

micade145 commented:
<img width="659" height="352" alt="image" src="https://github.com/user-attachments/assets/3c0d916e-2376-444d-aeaf-d97f6ad251e3" />

btw this is what I referenced for the INA241 to set a custom reference voltage. 

- for USB-C, I'll go ahead and add it; I can remove if it gets too cramped maybe

micade145 committed:
is this how you TVS

micade145 committed:
changed to TI generated filter

micade145 committed:
set reference voltage for current sense

micade145 committed:
scrutineering should be done, started on USB

micade145 committed:
copied USB-C from psom

micade145 committed:
something changed idk what though

micade145 commented:
@Rav4s @Lakshay983  Could I get another review pretty please :pray:

micade145 committed:
sch almost done

micade145 committed:
0603 footprints

micade145 committed:
dpdt switch for scrutineering

Lakshay983 reviewed (commented):
Whatever you use for the reference, use a reference IC instead of the divider since any noise in 3v3 will propagate to your reference
use smth like this:
https://www.ti.com/product-category/power-management/voltage-references/overview.html?utm_source=google&utm_medium=cpc&utm_campaign=app-lp-null-58700007987776466_Vref_overview_rsa-cpc-pp-google-ww_en_int&utm_content=Vref_overview&ds_k=ti+voltage+reference&gclsrc=aw.ds&gad_source=1&gad_campaignid=2051752292&gbraid=0AAAAAC068F0EkAFvM60z_9Sm4vtV0Agae&gclid=Cj0KCQjw5c_FBhDJARIsAIcmHK9RtmJf_PL3qevwSdlv6P9fOvsvxDEpmVIt6yIZOsrK9PxkIn2iMA0aAsbpEALw_wcB

<img width="741" height="227" alt="Image" src="https://github.com/user-attachments/assets/94be3589-077a-435e-bdd8-6c40aab1d176" />. 
why use an isolated DC-DC to power the microcontroller? The input 24V should be referenced to the same GND as the MCU. You can take the LDO and buck from psom for 24 - 5V and then 5 to 3v3 unisolated.

Use a DPDT switch for scrutineering so you can swap both the gnd reference and positive during scrutineering

Add bypass caps to all your ICs, there should be a section for it in each of their datasheets (your amplifiers, op amps, and ina, etc)

<img width="658" height="342" alt="Image" src="https://github.com/user-attachments/assets/c9cc428e-8816-47e8-b085-b5df616447c5" />

For the bootloader can you add an OR gate for setting the boot0 pin, basically if either the switch is high or a GPIO pin is high then we'll boot from flash.

<img width="407" height="305" alt="Image" src="https://github.com/user-attachments/assets/ad195dc0-da94-48b4-9823-74fc1799d77d" />. 
GND pins A8 and and B8

Can you make the op amp on the output a 2 channel op amp with one of them being a unity gain buffer. Ask @Rav4s for the standard op amp.

Can you add a reverse polarity circuit to this, just copy the one from VCU. 

<img width="607" height="180" alt="Image" src="https://github.com/user-attachments/assets/cfadb7d9-f569-4ebe-ace2-2e0f1891c801" />. 
pls fix

micade145 committed:
24-5 buck from psom, OR gate for bootload. more sheet organization

micade145 committed:
op amp

micade145 committed:
changed to 2 channel op amp, stole RPP from PSOM

Rav4s commented:
Nice job! love the use of buses it looks super clean lol
<img width="1163" height="1073" alt="image" src="https://github.com/user-attachments/assets/134ad014-4191-4cc6-ac42-e0fe520f4345" />

 - PWR looks good
 - don't listen to @Lakshay983 abt grounding the USB pins A8 and B8 idk where he got that from what a goon amirite. the USB spec says to leave them isolated if unused
 - CAN good
 - Make sure to add a voltage reference here. TI prob makes a 1.25v ref IC
<img width="889" height="415" alt="image" src="https://github.com/user-attachments/assets/4fb37807-570a-497b-bf17-1b7d226e4dd7" />

- Can you write expected voltage range from scrutineering board and corresponding Scrut_Out?
<img width="748" height="814" alt="image" src="https://github.com/user-attachments/assets/634f6efc-0370-4f0f-8586-eee8c90acc0c" />

- Flip this to be an input
<img width="346" height="220" alt="image" src="https://github.com/user-attachments/assets/c2e9e788-a8c8-4e6a-9806-07248b1d0de1" />

- Put the 10k resistor on the output of the OR gate so it pullups either input. Also maybe have a TVS on BOOT0, can use the same one as the PP IMON
<img width="1272" height="638" alt="image" src="https://github.com/user-attachments/assets/dc09b483-63b9-4ed6-bc36-7c538430c067" />

micade145 committed:
added voltage reference, shunt model; addressed comments

micade145 committed:
cleaned up parts lib, correct INA ic listed

micade145 committed:
added vertical usb c

micade145 committed:
renamed, added OR gate part

micade145 commented:
I'm starting layout, any last schematic comments? @Rav4s @Lakshay983
(I addressed the earlier comments)

micade145 committed:
lay out lay out

micade145 committed:
changed shunt model, moved around folders; initial layout, routing soon

micade145 committed:
9/20 workday: some layout stuff ye ye

KadhirTheEngineer commented:
The schematic is very nice. 

A few notes on the Layout (pre routing):
1. You can rotate components 45 degrees if you wanna get a tighter layout, and save a bunch of space between components and shunt. 
2. Trace length between I_Out and the MCU pin may cause signal integrity issues. Consider moving them closer together. 
3. Will vertical USB-C be accessible from the battery box, or will it be covered? If it will be covered, I might want to look into the panel mount stuff.

micade145 committed:
better layout, started routing

micade145 committed:
board is smaller ? idk

micade145 committed:
more routing later :(

micade145 committed:
more routing

micade145 committed:
more routing

micade145 committed:
review council

micade145 committed:
silkscreen work

micade145 committed:
added models

micade145 committed:
i don't know what I changed

Rav4s commented:
nice job! a few things:
- add stiching vias between your front/back gnd planes for a clean return path. especially near your high-current ICs
<img width="1323" height="759" alt="image" src="https://github.com/user-attachments/assets/d22387a2-4a16-4d87-a1f9-406da4453ff8" />

- pad 2 is very sad cause its not getting filled :(
<img width="495" height="408" alt="image" src="https://github.com/user-attachments/assets/f7a5a5fd-b57b-418f-8b9a-46b1a3aca11f" />

- same for these, you can expand the zone and maybe play with clearances a bit
<img width="407" height="350" alt="image" src="https://github.com/user-attachments/assets/0b43c1b0-3b9a-46f2-adf1-0d28a58e24de" />

- you can prolly make this path a bit cleaner
<img width="416" height="294" alt="image" src="https://github.com/user-attachments/assets/d6116267-97fa-46ef-a49d-770272a5a0c1" />

- via in pad bad
<img width="379" height="382" alt="image" src="https://github.com/user-attachments/assets/2b712cae-701c-4b0e-b0a8-6cd1c3f1ab23" />

- put more vias to VBUS_IN
<img width="634" height="385" alt="image" src="https://github.com/user-attachments/assets/a6a13fbd-86f5-4c4f-9fee-e41ea4cf085f" />

- try to not have 3v3 in between here so the gnd thermal relief can fill + cleaner connection to caps return. maybe put the vias above the 3v3 pads instead of below
<img width="990" height="575" alt="image" src="https://github.com/user-attachments/assets/6335e47e-65c8-4789-9829-b78052dfffd2" />

- no teardrop on osc nets bc high frequency smth smth
<img width="797" height="574" alt="image" src="https://github.com/user-attachments/assets/12a62bb5-656f-4f20-9880-cbe6e37a5328" />

- put a grid of vias in the thermal pad (25). similar to what u did with RTN
<img width="915" height="907" alt="image" src="https://github.com/user-attachments/assets/09d21c30-5137-480d-8eed-bd3488c40aaa" />

- put more vias for the 3.3V_HV
<img width="856" height="699" alt="image" src="https://github.com/user-attachments/assets/0c6d8030-a6ef-4bda-a66e-0123efed19ab" />

- via in pad!!!
<img width="980" height="882" alt="image" src="https://github.com/user-attachments/assets/70d564ad-7c23-4540-8c5c-4050a03037db" />

 - label the scrut and shunt + and - pins here, will make testing easier
<img width="1144" height="972" alt="image" src="https://github.com/user-attachments/assets/02180ce8-74fa-42f8-b16f-53db4990762e" />

- add the remaining 3d models

Rav4s commented:
also round corners pls

micade145 committed:
more silkscreen, addressed some comments

micade145 committed:
more comments addressed

micade145 committed:
drc drc

micade145 committed:
updated some component part nums

Lakshay983 reviewed (changes requested):
LGTM other than a few notes:

put testpoint on sense+ and OUT+

Can you put month and year on silkscreen of PCB

label pinout for peripheral CAN connector

what is via on boot0 doing?

run tools -> Cleanup track and vias

add shunt 3D model

Looks great!

micade145 committed:
new dpdt switch, layout done

micade145 committed:
generated production stuff

micade145 committed:
updated production files again

micade145 committed:
added a 3d model nothing too crazy i think

micade145 committed:
updated step file

micade145 committed:
docs, bom pushed

micade145 committed:
added bom

micade145 committed:
bom again

micade145 committed:
updated amperes docs
